src/polymer/elements/paper-toolbar-v1-to-v2.ts

```diff
@@ -0,0 +1,209 @@
+            'content or `slot="middle"` / `slot="bottom"` to distribute to ' +
+            'the middle or bottom bar.',
+          severity: Severity.WARNING,
+          sourceRange: startTagSourceRange
+        });
+
+        let desiredSlot = 'top';
+        if (dom5.hasSpaceSeparatedAttrValue('class', 'middle')(node)) {
+          desiredSlot = 'middle';
+        }
+        if (dom5.hasSpaceSeparatedAttrValue('class', 'bottom')(node)) {
+          desiredSlot = 'bottom';
+        }
+
+        const [startOffset, endOffset]
+          = parsedDocument.sourceRangeToOffsets(startTagSourceRange);
+        const startTagText =
+          parsedDocument.contents.slice(startOffset, endOffset);
+        const isSelfClosing = startTagText.endsWith('/>');
+
+        warning.fix = [{
+          range: startTagSourceRange,
+          replacementText:
+            startTagText.slice(0, isSelfClosing ? -2 : -1) +
+            ` slot="${desiredSlot}"` +
+            (isSelfClosing ? '/' : '') + '>',
+        }];
+
+        warnings.push(warning);
+      }
+
+      // Non-whitespace-only text nodes, which were previously distributed into
+      // a default slot, now need to be wrapped in `<span slot="top">...</span>`.
+      if (node.nodeName === '#text' && node.value!.trim() !== '') {
+        const textNodeSourceRange = parsedDocument.sourceRangeForNode(node)!;
+
+        const warning = new FixableWarning({
+          parsedDocument,
+          code: this.code,
+          message: '<paper-toolbar> no longer has a default slot: this ' +
+            'text node will not appear in the composed tree. Wrap with ' +
+            '`<span slot="top">...</span>` to distribute to the same ' +
+            'position as the previous default content.',
+          severity: Severity.WARNING,
+          sourceRange: textNodeSourceRange
+        });
+
+        const fullText = node.value!;
+        const trimmedText = fullText.trim();
+        const trimmedOffset = fullText.indexOf(trimmedText);
+
+        const replacementText =
+          // leading whitespace:
+          fullText.substring(0, trimmedOffset) +
+          // wrapper and trimmed text:
+          '<span slot="top">' + trimmedText + '</span>' +
+          // trailing whitespace:
+          fullText.substring(trimmedOffset + trimmedText.length);
+
+        warning.fix = [{
+          range: textNodeSourceRange,
+          replacementText,
+        }];
+
+        warnings.push(warning);
+      }
+    };
+
+    for (const paperToolbar of paperToolbars) {
+      let suspectNodes = Array.from(paperToolbar.childNodes!);
+
+      const nodeIsTemplateExtension = (node: dom5.Node) => {
+        if (!node.attrs) return false;
+
+        const isAttr = node.attrs.find(attr => attr.name === 'is');
+        return (
+          node.tagName === 'template' &&
+          isAttr &&
+          ['dom-bind', 'dom-if', 'dom-repeat'].includes(isAttr.value)
```

Comment:
Should `'dom-template'` be in here? What does it do?

Reply:
Just talked with @sorvell, sounds like these three are the only ones we care about. `dom-template` isn't in Polymer 2.

Reply:
After further discussion, it sounds like`dom-template` is just a more controllable `dom-bind`, so I'll need to add it here too.